test/topology_custom/test_repair.py

```diff
@@ -67,3 +70,77 @@ async def test_enable_compacting_data_for_streaming_and_repair_live_update(manag
     # After the update to the config above, the next repair should pick up the updated value.
     await manager.api.repair(node1.ip_addr, "ks", "tbl")
     assert (await get_injection_params(manager, node1.ip_addr, "maybe_compact_for_streaming"))["compaction_enabled"] == "true"
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tombstone_gc_for_streaming_and_repair_compaction(manager):
+    """
+    Check that:
+    * enable_tombstone_gc_for_streaming_and_repair_compaction=1 works as expected
+    * enable_tombstone_gc_for_streaming_and_repair_compaction=0 works as expected
+    * enable_tombstone_gc_for_streaming_and_repair_compaction is live-update
+    """
+    cmdline = [
+            "--enable-compacting-data-for-streaming-and-repair", "1",
+            "--enable-tombstone-gc-for-streaming-and-repair-compaction", "1",
+            "--enable-cache", "0",
+            "--hinted-handoff-enabled", "0",
+            "--smp", "1",
+            "--logger-log-level", "api=trace:database=trace"]
+    node1 = await manager.server_add(cmdline=cmdline)
+    node2 = await manager.server_add(cmdline=cmdline)
+
+    cql = manager.get_cql()
+
+    cql.execute("CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 2}")
+    cql.execute("CREATE TABLE ks.tbl (pk int, ck int, PRIMARY KEY (pk, ck)) WITH compaction = {'class': 'NullCompactionStrategy'}")
+
+    await manager.server_stop_gracefully(node2.server_id)
+
+    stmt = SimpleStatement("DELETE FROM ks.tbl WHERE pk = 0 AND ck = 0", consistency_level=ConsistencyLevel.ONE)
+    cql.execute(stmt)
+
+    await manager.server_start(node2.server_id, wait_others=1)
+
+    # Flush memtables and remove commitlog, so we can freely GC tombstones.
+    await manager.server_restart(node1.server_id, wait_others=1)
```

Comment:
Nitpick: don't we have a REST API for flushing memtables instead of restarting the node? :-(

Reply:
Simple flush left commitlog behind, and tombstone GC wouldn't work, hence I resorted to restarting the node.